### PR TITLE
[gdal] add feature 'jpegxl'

### LINKED ISSUE #29785: [gdal] Enable JPEG-XL driver via libjxl
GDAL supports the JPEG-XL format via libjxl (https://gdal.org/drivers/raster/jpegxl.html). libjxl is already in vcpkg thus it should be straightforward to add this feature to the GDAL port.

### PR DESCRIPTION
**Pull request by jimwang118**
Fixes https://github.com/microsoft/vcpkg/issues/29785

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

All feature passed with following triplets:

- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/gdal/portfile.cmake [hunk 1]
```diff
@@ -55,6 +55,7 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         core             GDAL_USE_ZLIB
         zstd             GDAL_USE_ZSTD
         tools            BUILD_APPS
+        jpegxl           GDAL_USE_JXL
```
Comment (dg0yt) [inline ports/gdal/portfile.cmake:58]: This would need to go before `lerc`. The driver options are sorted by `GDAL_USE_<DRIVER>`.

**Review by dg0yt — COMMENTED** [review]
As usual, this needs more work to deal with static link libraries needed for using libjxl. CMake config for libjxl is coming but not yet available.
It might also be resolved with pkg-config but it needs patching.
That's why I didn't add this feature yet. But I do check open issues every time I do a version update, and I try to upstream changes if needed. In fact, I even add some comments to highway which is a dependency of libjxl.

**Comment by JonLiu1993** [comment]
@dg0yt, I saw your question about this PR above, you can review it again to see if it is reasonable? Thanks!

**Comment by dg0yt** [comment]
Did you already test building with the feature and processing a JXL image?
Since yesterday, I'm even less convinced that this it is worth spending my time on testing this PR:
https://github.com/microsoft/vcpkg/issues/29785#issuecomment-1455128845

**Pull request closed without merging**